### PR TITLE
refactor: Migrate HeaderActionsDropdown to typescript

### PR DESCRIPTION
**Pull request by EnxDev**
### SUMMARY
Migrate HeaderActionsDropdown to typescript
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N.A
### TESTING INSTRUCTIONS
- All tests should pass
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/index.tsx [hunk 1]
```diff
@@ -35,97 +34,60 @@ import FilterScopeModal from 'src/dashboard/components/filterscope/FilterScopeMo
...
 
-  changeRefreshInterval(refreshInterval, isPersistent) {
+  changeRefreshInterval = (refreshInterval: number, isPersistent: boolean) => {
     this.props.setRefreshFrequency(refreshInterval, isPersistent);
     this.props.startPeriodicRender(refreshInterval * 1000);
-  }
+  };
 
-  handleMenuClick({ key }) {
+  handleMenuClick = ({ key }: Record<string, any>) => {
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/index.tsx:90]: nit: shouldn't the type here be sth like `{ key: MenuKeys }`?

**Review by kgabryje — APPROVED** [review]
Looks great, thank you for your work! Added 1 minor comment

File: superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/index.tsx (hunk 1 above)
Comment (EnxDev) [reply superset-frontend/src/dashboard/components/Header/HeaderActionsDropdown/index.tsx:90]: { key: MenuKeys } doesn't work. 

I believe the appropriate type is { MenuInfo } from 'rc-menu/lib/interface';  however, rc-menu is not listed in the project's dependencies.

**Pull request merged**